user/src/clients/postgres.ts

```diff
@@ -0,0 +1,44 @@
+import { Pool, QueryResult, QueryResultRow } from 'pg';
+
+class PostgresClient {
+    private readonly pool: Pool;
+
+    constructor(connectionString: string) {
+        this.pool = new Pool({
+            max: process.env.POSTGRES_CONN_COUNT ? parseInt(process.env.POSTGRES_CONN_COUNT) : undefined,
+            connectionString: connectionString,
+            idleTimeoutMillis: process.env.POSTGRES_TIMEOUT ? parseInt(process.env.POSTGRES_TIMEOUT) : undefined
+        });
+    }
+
+    public async query<T extends QueryResultRow>(
+        sql: string,
+        values?: any[]
+    ): Promise<QueryResult<T>> {
+        const client = await this.pool.connect();
+
+        try {
+            return await client.query(sql, values);
```

Comment:
Not sure if GCP will have a similar issue, but I had done this with AWS before - the lambda function stops execution upon `return` which left an open db connection. We can monitor if this would happen for GCP.